bk-api/bk_api.py

```diff
@@ -1077,6 +1117,15 @@ def post(self, node_id):
 
             return jsonify(result)
 
+        if "vsn" in postData:
```

Comment:
i think we want to do this before `deploy_wes` , how do ensure that?

Reply:
Ensure what? It just needs to call that endpoint and it will get the VSN from the node. Very similar to the deploy_wes service.

Reply:
when a node registers with beekeeper the steps that are taken are 
- `deploy_wes`
- `vsn`
how do you ensure the `vsn` step is done before `deploy_wes` on node registration?

Reply:
That is done externally by the bk-deploy-manager for deploy_wes, but VSN is not added there yet. 
Not sure where you guys want to add it because we need to first answer these questions (my comment from yesterday):
```
There are two type of actions that we need with VSN:
The first one is to check that VSN is present in the node -> check that vsn is null and grab the vsn on the node to update the null value
The second one is to check that the VSN is up-to-date. -> I get the current VSN in the db and check against the nodes vsn.
```

Reply:
Lets talk about this as a team.  as I don't remember why we wouldn't just do this on Node registration.  as the VSN doesn't change very often at all.  and any manual changes of VSN (which I only do very very rarely) would require me to just do a new `curl localhost:5000/node/0000000000000001 -d '{"vsn": true}'`

Reply:
I agree. I think we should make this a registration time thing.

My own preference is generally to not "reach into the node", if possible. That would also allow more generic testing of registration, since a test program can provide all that info, even when there's no real node backing it.

Reply:
I agree. I think we should make this a registration time thing.

My own preference is generally to not "reach into the node", if possible. That would also allow more generic testing of registration, since a test program can provide all that info, even when there's no real node backing it.